### DISCUSSION TITLE
mac 编译失败

### DISCUSSION THREAD

**Discussion by ChengChen0301** [sjtug/SJTUThesis#902] [提问]
下载模板包后，终端使用make all 进行编译，跳出以下错误信息。之前没用过latexmk，所以不太能看懂，求解决。overleaf上面的模板能用，但是由于图片太多，无法编译全文，只能分章节编译，所以迫切需要使用本地版本编译全文。

Latexmk: Initialization file '.latexmkrc' gave an error:
     Undefined subroutine &main::ensure_path called at (eval 8) line 10, <GEN0> chunk 1.

Latexmk: Stopping because of problem with rc file

**Comment by LogCreative** [comment]
请给出 MacTeX（或 TeX Live）的版本、latexmk 的版本。

**Reply by ChengChen0301** [reply]
谢谢，重新安装了MacTeX，可以编译了！

**Comment by AlexaraWu** [comment]
报错提示子程序 `ensure_path` 未定义，推测你的 `latexmk` 版本过低。配置文件需要 `latexmk` 4.59 或以上版本。

建议安装较新的 TeX 发行版。

**Reply by ChengChen0301** [reply]
谢谢，更新了最新版的MacTex，已解决。
